app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/snippet/SnippetListDetailComposable.kt

```diff
@@ -0,0 +1,118 @@
+package com.bottlerocketstudios.brarchitecture.ui.snippet
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.res.stringResource
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.compose.composable
+import com.bottlerocketstudios.brarchitecture.R
+import com.bottlerocketstudios.brarchitecture.ui.ComposeActivity
+import com.bottlerocketstudios.brarchitecture.ui.Routes
+import com.bottlerocketstudios.compose.snippets.CreateSnippetScreen
+import com.bottlerocketstudios.compose.snippets.SnippetUiModel
+import com.bottlerocketstudios.compose.snippets.SnippetsBrowserScreen
+import com.bottlerocketstudios.compose.snippets.SnippetsBrowserScreenState
+import com.bottlerocketstudios.compose.util.LaunchCollection
+import com.bottlerocketstudios.compose.util.asMutableState
+import com.bottlerocketstudios.compose.util.toStringIdHelper
+import com.bottlerocketstudios.compose.widgets.listdetail.AnimatedListDetail
+import kotlinx.coroutines.launch
+import org.koin.androidx.viewmodel.ext.android.getViewModel
+
+// Used to represent create item in list so it can be "selected" for detail view
+private val CreateSnippetItem = SnippetUiModel(
+    id = "CREATE_SNIPPET_SCREEN",
+    workspaceSlug = "",
+    title = "CREATE_SNIPPET_SCREEN",
+    userName = "UI_MODEL",
+    formattedLastUpdatedTime = "".toStringIdHelper()
+)
+
+// TODO - Custom Animations for entrance of detail.   Try to use same animation spec for navigation side and visibility
+fun ComposeActivity.newSnippetsComposable(navGraphBuilder: NavGraphBuilder) {
+    navGraphBuilder.composable(Routes.NewSnippet) {
+        val scope = rememberCoroutineScope()
+        val snippetsViewModel: SnippetsViewModel = getViewModel()
+        val config = LocalConfiguration.current
+
+        val list = snippetsViewModel.snippets.collectAsState(initial = emptyList())
+        controls.title = stringResource(id = R.string.snippets_title)
+        controls.topLevel = true
+
+        AnimatedListDetail(
+            list = list.value + CreateSnippetItem ,
+            keyProvider = { it.id },
+            smallScreen = config.smallestScreenWidthDp < 580) {
```

Comment:
I'll look into window size class usage.

Reply:
Cool! I see you've added the enum to represent the different types. Last thing I see for this PR is to add logic to determine the current window type and then use it instead of the isSmallScreen boolean